R/merge_lists.R

```diff
@@ -0,0 +1,58 @@
+#' Merge two lists
+#'
+#' This function takes two lists (`base` and `overlay`), and merges them (by
+#' default recursively) into a single list. If a key is present in `overlay`,
+#' it is inherited from `overlay`, but keys missing in `overlay` will be
+#' inherited from `base`
+#'
+#' Code is heavily taken from the `merge` function in the `config` package.
+#' Notable differences between the two functions are:
+#' - This does not reorder keys in the lists
+#'
+#' @param base_list a named list
+#' @param overlay_list a named list
+#' @param recursive Should list be merged recurisvely, or only with top level
+#' keys?
+#'
+#' @return nested list of file details, length the same as the input vector.
```

Comment:
NB: Is it necessarily "file details"? Seems like this function just handles arbitrary lists